omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,331 @@
+		
+		Assert.assertFalse(diagnosis.isVoided());
+		Assert.assertNull(diagnosis.getDateVoided());
+		Assert.assertNull(diagnosis.getVoidedBy());
+		Assert.assertNull(diagnosis.getVoidReason());
+		
+	}
+	
+	@Test
+	public void shouldUpdateANonCodedDiagnosis() throws Exception {
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(RestTestConstants2_2.UPDATABLE_NON_CODED_DIAGNOSIS_UUID);
+		
+		Assert.assertFalse(diagnosis.getVoided());
+		Assert.assertNull(diagnosis.getDiagnosis().getCoded());
+		Assert.assertEquals((int) diagnosis.getRank(), 2);
+		Assert.assertEquals(diagnosis.getCondition().getUuid(), "e804ee60-ecbc-4d70-abda-1e4f6f64e5b5");
+		
+		SimpleObject codedOrFreeText = new SimpleObject();
+		
+		codedOrFreeText.add("coded", concept.getUuid());
+		codedOrFreeText.add("specificName", conceptName.getUuid());
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("voided", true);
+		attributes.add("diagnosis", codedOrFreeText);
+		attributes.add("condition", condition.getUuid());
+		attributes.add("rank", 1);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/"
+		        + RestTestConstants2_2.UPDATABLE_NON_CODED_DIAGNOSIS_UUID);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		Diagnosis new_diagnosis = diagnosisService
+		        .getDiagnosisByUuid(RestTestConstants2_2.UPDATABLE_NON_CODED_DIAGNOSIS_UUID);
+		
+		Assert.assertTrue(new_diagnosis.getVoided());
+		Assert.assertEquals(new_diagnosis.getDiagnosis().getCoded().getUuid(), concept.getUuid());
+		Assert.assertEquals((int) new_diagnosis.getRank(), 1);
+		Assert.assertEquals(new_diagnosis.getCondition().getUuid(), condition.getUuid());
+	}
+	
+	@Test
+	public void shouldUpdateACodedDiagnosis() throws Exception {
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(RestTestConstants2_2.UPDATABLE_CODED_DIAGNOSIS_UUID);
+		
+		Assert.assertTrue(diagnosis.getVoided());
+		Assert.assertNull(diagnosis.getDiagnosis().getNonCoded());
+		Assert.assertEquals((int) diagnosis.getRank(), 1);
+		Assert.assertEquals(diagnosis.getCondition().getUuid(), "e804ee60-ecbc-4d70-abda-1e4f6f64e5b5");
+		
+		SimpleObject codedOrFreeText = new SimpleObject();
+		
+		codedOrFreeText.add("nonCoded", "Some condition");
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("voided", false);
+		attributes.add("diagnosis", codedOrFreeText);
+		attributes.add("condition", condition.getUuid());
+		attributes.add("rank", 2);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/"
+		        + RestTestConstants2_2.UPDATABLE_CODED_DIAGNOSIS_UUID);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		Diagnosis new_diagnosis = diagnosisService.getDiagnosisByUuid(RestTestConstants2_2.UPDATABLE_CODED_DIAGNOSIS_UUID);
```

Comment:
Did you take a look at our naming conventions? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-NamingConventions

Reply:
Yeah, I did. This was an overlook. Let me fix it